lib/puppet/util/windows/api_types.rb

```diff
@@ -263,6 +266,81 @@ class FILETIME < FFI::Struct
              :dwHighDateTime, :dword
     end
 
+    # https://docs.microsoft.com/en-us/previous-versions/windows/desktop/legacy/aa379560(v=vs.85)
+    # typedef struct _SECURITY_ATTRIBUTES {
+    #   DWORD  nLength;
+    #   LPVOID lpSecurityDescriptor;
+    #   BOOL   bInheritHandle;
+    # } SECURITY_ATTRIBUTES, *PSECURITY_ATTRIBUTES, *LPSECURITY_ATTRIBUTES;
+    class SECURITY_ATTRIBUTES < FFI::Struct
+      layout(
+        :nLength, :dword,
+        :lpSecurityDescriptor, :pointer,
+        :bInheritHandle, :int
+      )
+    end
+
+    # sizeof(STARTUPINFO) == 68
+    # https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/ns-processthreadsapi-startupinfoa
+    # typedef struct _STARTUPINFOA {
+    #   DWORD  cb;
+    #   LPSTR  lpReserved;
+    #   LPSTR  lpDesktop;
+    #   LPSTR  lpTitle;
+    #   DWORD  dwX;
+    #   DWORD  dwY;
+    #   DWORD  dwXSize;
+    #   DWORD  dwYSize;
+    #   DWORD  dwXCountChars;
+    #   DWORD  dwYCountChars;
+    #   DWORD  dwFillAttribute;
+    #   DWORD  dwFlags;
+    #   WORD   wShowWindow;
+    #   WORD   cbReserved2;
+    #   LPBYTE lpReserved2;
+    #   HANDLE hStdInput;
+    #   HANDLE hStdOutput;
+    #   HANDLE hStdError;
+    # } STARTUPINFOA, *LPSTARTUPINFOA;
+    class STARTUPINFO < FFI::Struct
+      layout(
+        :cb, :ulong,
+        :lpReserved, :string,
+        :lpDesktop, :string,
+        :lpTitle, :string,
+        :dwX, :dword,
+        :dwY, :dword,
+        :dwXSize, :dword,
+        :dwYSize, :dword,
+        :dwXCountChars, :dword,
+        :dwYCountChars, :dword,
+        :dwFillAttribute, :dword,
+        :dwFlags, :dword,
+        :wShowWindow, :word,
+        :cbReserved2, :word,
+        :lpReserved2, :pointer,
+        :hStdInput, :handle,
+        :hStdOutput, :handle,
+        :hStdError, :handle
+      )
+    end
+
+    # https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/ns-processthreadsapi-process_information
+    # typedef struct _PROCESS_INFORMATION {
+    #   HANDLE hProcess;
+    #   HANDLE hThread;
+    #   DWORD  dwProcessId;
+    #   DWORD  dwThreadId;
+    # } PROCESS_INFORMATION, *PPROCESS_INFORMATION, *LPPROCESS_INFORMATION;
+    class PROCESS_INFORMATION < FFI::Struct
+      layout(
+        :hProcess, :handle,
+        :hThread, :handle,
+        :dwProcessId, :ulong,
+        :dwThreadId, :ulong
```

Comment:
migrated this from win32-process, but the struct definition is using `DWORD`, maybe this should be `:dword` and drop `ulong` type definition